### ISSUE TITLE
Unable to compile on windows

### ISSUE THREAD

**Issue by viperscape** [sfackler/rust-openssl#241]
I'm compiling openssl-win32 on windows with latest rust nightly. 
Primary error seems to be: 
``` 
C:\OpenSSL-Win32/include/openssl/sha.h:184:16: error: expected ':', ',', ';', '}
' or '__attribute__' before 'h'

     SHA_LONG64 h[8]; 
```

Any help is appreciated, sorry if I missed something obvious! (Do I need openssl-win64 instead?)

More detailed error:
```
...
error: aborting due to 851 previous errors
failed to run custom build command for `openssl-sys v0.6.4`
Process didn't exit successfully: `C:\users\Chris\sire\target\debug\build\openss
l-sys-f89b3551681c09fc\build-script-build` (exit code: 101)
--- stdout
cargo:rustc-link-search=native=C:\OpenSSL-Win32
cargo:rustc-link-lib=dylib=eay32
cargo:rustc-link-lib=dylib=ssl32
TARGET = Some("i686-pc-windows-gnu")
TARGET = Some("i686-pc-windows-gnu")
CARGO_MANIFEST_DIR = Some("C:\\cygwin64\\home\\Chris\\.cargo\\registry\\src\\git
hub.com-1285ae84e5963aae\\openssl-sys-0.6.4")
OUT_DIR = Some("C:\\users\\Chris\\sire\\target\\debug\\build\\openssl-sys-f89b35
51681c09fc\\out")
OPT_LEVEL = Some("0")
PROFILE = Some("debug")
debug 0
TARGET = Some("i686-pc-windows-gnu")
HOST = Some("i686-pc-windows-gnu")
CC_i686-pc-windows-gnu = None
CC_i686_pc_windows_gnu = None
HOST_CC = None
CC = None
TARGET = Some("i686-pc-windows-gnu")
HOST = Some("i686-pc-windows-gnu")
CFLAGS_i686-pc-windows-gnu = None
CFLAGS_i686_pc_windows_gnu = None
HOST_CFLAGS = None
CFLAGS = None
TARGET = Some("i686-pc-windows-gnu")
HOST = Some("i686-pc-windows-gnu")
CC_i686-pc-windows-gnu = None
CC_i686_pc_windows_gnu = None
HOST_CC = None
CC = None
running: "gcc" "-O0" "-c" "-ffunction-sections" "-fdata-sections" "-mwin32" "-m3
2" "-I" "C:\OpenSSL-Win32/include" "C:\cygwin64\home\Chris\.cargo\registry\src\g
ithub.com-1285ae84e5963aae\openssl-sys-0.6.4\src/openssl_shim.c" "-o" "C:\users\
Chris\sire\target\debug\build\openssl-sys-f89b3551681c09fc\out\src\openssl_shim.
o"


command did not execute successfully, got: exit code: 1



--- stderr
In file included from C:\OpenSSL-Win32/include/openssl/x509.h:107:0,
                 from C:\OpenSSL-Win32/include/openssl/ssl.h:156,
                 from C:\cygwin64\home\Chris\.cargo\registry\src\github.com-1285
ae84e5963aae\openssl-sys-0.6.4\src/openssl_shim.c:2:
C:\OpenSSL-Win32/include/openssl/sha.h:184:16: error: expected ':', ',', ';', '}
' or '__attribute__' before 'h'
     SHA_LONG64 h[8];
                ^
thread '<main>' panicked at 'explicit panic', C:\cygwin64\home\Chris\.cargo\regi
stry\src\github.com-1285ae84e5963aae\gcc-0.3.8\src\lib.rs:497
```

**Comment by sfackler** [comment]
That's a really bizarre error. What version of GCC are you using?

**Comment by viperscape** [comment]
```
$ gcc --version
gcc (GCC) 4.9.3

$ rustc --version
rustc 1.2.0-nightly (0cc99f9cc 2015-05-17) (built 2015-05-18)

$ cargo --version
cargo 0.2.0-nightly (ac61996 2015-05-17) (built 2015-05-18)
```

**Comment by semmaz** [comment]
I think that's the same issue cybergeek94 have [here](https://github.com/hyperium/hyper/issues/607#issuecomment-120852518) (see my response).

**Comment by viperscape** [comment]
Feel free to merge the two issues, they seem similar enough now. I am using cygwin's gcc, so its an emulated shell environment, but my paths are set up properly to my knowledge. I'm actually compiling this from a bash shell in windows. Maybe I'll try mingw's gcc, I think I can make that change in cygwin.

```
$ which gcc
/usr/bin/gcc
```

**Comment by viperscape** [comment]
I think I fixed it, I uninstalled gcc and installed the mingw-gcc associated binaries (which is required by quite a few other packages, so some were technically installed already). Problem is that they are under a long name other than gcc.exe-- I copied one and renamed as that and it compiled. I'm not sure why the regular gcc doesn't work, but that's a work around. Now I just need to figure out how to compile winapi

Thanks
